pkg/issuer/acme/issue.go

```diff
@@ -426,7 +417,6 @@ func orderLabels(crt *v1alpha1.Certificate) map[string]string {
 	for k, v := range crt.Labels {
 		lbls[k] = v
 	}
-	lbls[certificateNameLabelKey] = crt.Name
```

Comment:
@munnerz I was wondering whether to update `buildOrder` to include the annotation with the secret name, since given the change there is no trace of the cert name in the order anymore. But it seems to be unnecessary so far.

If you prefer to add it, I was thinking of sticking 
```go
Annotations:          orderAnnotations(crt),
```

with 

```go
func orderAnnotations(crt *v1alpha1.Certificate) map[string]string {
       return map[string]string{
               v1alpha1.CertificateNameKey: crt.Name,
       }
}
```

at L410. What do you think?